pkg/access/commands/use.go

```diff
@@ -85,47 +91,43 @@ func AccessDetails() *cobra.Command {
 
 	flags := cmd.Flags()
 	flags.StringVarP(&outputFormat, outputFlagName, "o", "instructions", "output format")
-	flags.BoolVarP(&shouldExecuteAccessCommand, runFlagName, "r", false, "output format")
+	flags.BoolVarP(&shouldExecuteAccessCommand, runFlagName, "r", false, "execute the cli command")
 
 	return cmd
 }
 
-func verifyOutputFormatIsSupported(ctx context.Context, client *aponoapi.AponoClient, id string, outputFormat string) error {
-	session, _, err := client.ClientAPI.AccessSessionsAPI.GetAccessSession(ctx, id).Execute()
-	if err != nil {
-		return fmt.Errorf("access session with id %s not found", id)
-	}
-
-	for _, supportedFormat := range session.ConnectionMethods {
-		if supportedFormat == outputFormat {
-			return nil
-		}
-	}
-
-	return fmt.Errorf("unsupported output format: %s. use one of: %s", outputFormat, strings.Join(session.ConnectionMethods, ", "))
-}
-
-func executeAccessDetails(ctx context.Context, client *aponoapi.AponoClient, accessID string) error {
+func executeAccessDetails(cobraCmd *cobra.Command, client *aponoapi.AponoClient, session *clientapi.AccessSessionClientModel) error {
 	if runtime.GOOS == "windows" {
 		return errors.New("--run flag is not supported on windows")
 	}
 
-	err := verifyOutputFormatIsSupported(ctx, client, accessID, cliOutputFormat)
+	if !slices.Contains(session.ConnectionMethods, cliOutputFormat) {
+		return fmt.Errorf("--run flag is not supported for session id %s", session.Id)
+	}
+
+	accessDetails, _, err := client.ClientAPI.AccessSessionsAPI.GetAccessSessionAccessDetails(cobraCmd.Context(), session.Id).Execute()
 	if err != nil {
-		return fmt.Errorf("--run flag is not supported for session id %s", accessID)
+		return fmt.Errorf("error getting access details for session id %s: %w", session.Id, err)
 	}
 
-	accessDetails, _, err := client.ClientAPI.AccessSessionsAPI.GetAccessSessionAccessDetails(ctx, accessID).Execute()
+	err = executeCommand(cobraCmd, accessDetails.GetCli())
 	if err != nil {
-		return fmt.Errorf("error getting access details for session id %s: %w", accessID, err)
+		return err
 	}
 
-	cmd := exec.CommandContext(ctx, "sh", "-c", accessDetails.GetCli()) //nolint:gosec // This is a command that should be executed for the user
+	return nil
+}
+
+func executeCommand(cobraCmd *cobra.Command, command string) error {
 	var stderr bytes.Buffer
+	cmd := exec.CommandContext(cobraCmd.Context(), "sh", "-c", command)
+	cmd.Stdout = cobraCmd.OutOrStdout()
+	cmd.Stdin = cobraCmd.InOrStdin()
```

Comment:
this solves the pgcli?

Reply:
yes (not only pgcli, also mongosh and probably others